### PR TITLE
Adds BIG-IP integration to WAAgent

### PR DESCRIPTION
**Pull request by caphrim007**
This patch includes the implementation portion of BIG-IP integration
with WAAgent.

### CODE REVIEW

**Comment by caphrim007** [comment]
@brendandixon ready for review

**Comment by brendandixon** [comment]
@caphrim007 I made one minor comment, but if you're fine with this we can merge.

**Comment by caphrim007** [comment]
@brendandixon i'm not entirely sure. on AzureStack is shows up as cdrom, but in Azure Public Cloud it was showing up as sr**

So that's why I left both of them.

**Comment by caphrim007** [comment]
@brendandixon this pr is certainly not the last one, related to bigip, that i'll send upstream. i fully expect more things to need to be fixed as we move towards parity between products.

File: azurelinuxagent/common/osutil/bigip.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        Note that this method will return upon the first device found, but in my
+        tests with 12.1.1 it will also find /dev/sr0 on occasion. This is NOT the
+        correct CD/DVD device though.
+
+        :todo: Consider just always returning "/dev/cdrom" here if that device device
+               exists on all platforms that are supported on Azure(Stack)
+        :param dev_dir: The root directory from which to look for devices
+        """
+        patten = r'(sr[0-9]|hd[c-z]|cdrom[0-9]?)'
+        for dvd in [re.match(patten, dev) for dev in os.listdir(dev_dir)]:
```
Comment (brendandixon) [inline azurelinuxagent/common/osutil/bigip.py]: @caphrim007 All good then.

**Pull request merged**